kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/checksum/MetadataChecksumGenerator.java

```diff
@@ -27,6 +34,13 @@ static String checksumFor(HasMetadata... metadataSources) {
         return checksum.encode();
     }
 
+    static String checksumFor(Collection<HasMetadata> metadataSources) {
+        HasMetadata[] array = metadataSources.stream()
+                .sorted(Comparator.comparing(ResourcesUtil::toLocalRef))
```

Comment:
maybe we should take a List and leave the sorting to the client

Reply:
have backed off having this generator handle the ordering as sometimes the client will know a deterministic order without having to re-sort. Also there could be some risks to sorting based on the LocalRef if somehow duplicates were present in the input.

Reply:
It does feel like a nice property to work on in future though, to be able to hand all referents to the checksummer and have the same checksum pop out regardless of input order etc.